tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -347,6 +350,60 @@ void SetHtmlResponseContent()
             return blockingAction;
         }
 
+        private string GetJsonResponse()
+        {
+            if (_blockedJsonTemplateCache != null)
+            {
+                return _blockedJsonTemplateCache;
+            }
+
+            if (_settings.BlockedJsonTemplatePath != null && AppDomain.CurrentDomain.BaseDirectory != null)
+            {
+                var fullPath =
+                    Path.IsPathRooted(_settings.BlockedJsonTemplatePath)
+                        ? _settings.BlockedJsonTemplatePath
+                        : Path.Combine(AppDomain.CurrentDomain.BaseDirectory, _settings.BlockedJsonTemplatePath);
+                if (File.Exists(fullPath))
+                {
+                    _blockedJsonTemplateCache = File.ReadAllText(fullPath);
```

Comment:
NIT: While it's checked that the file exists, potentially some other exceptions, like UnauthorizedAccessException could be launched. You might want to catch them here?

Reply:
actually, I think it's than a nit, an exception here could lead to serious misbehaviour, so definitely a try catch is needed. See 4f7baf801